### PR TITLE
Modifying test suite

### PR DESCRIPTION
**Pull request by ISauve**
*Note: Please remember to review the Datadog [Contribution Guidelines](https://github.com/DataDog/dd-agent/blob/master/CONTRIBUTING.md)
if you have not yet done so.*

### What does this PR do?

Modifies the test suite so that it test everything properly after the nozzle restructuring in https://github.com/DataDog/datadog-firehose-nozzle/pull/18

### Motivation

Needed to be done after that PR!

### Testing Guidelines

An overview on [testing](https://github.com/DataDog/dd-agent/blob/master/tests/README.md)
is available in our contribution guidelines.

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

**Comment by ISauve** [comment]
[Use channels instead of locks in the nozzle](https://trello.com/c/fttUtRlK/416-use-channels-instead-of-locks-in-the-nozzle)

File: datadogclient/datadog_client_test.go [hunk 1]
```diff
@@ -175,623 +153,237 @@ var _ = Describe("DatadogClient", func() {
...
-		err = json.Unmarshal(bodies[1], &payload)
-		Expect(err).NotTo(HaveOccurred())
+		f := func() int {
+			return len(bodies)
+		}
 
-		errMetric = findSlowConsumerMetric(payload)
-		Expect(findSlowConsumerMetric(payload)).ToNot(BeNil())
-		Expect(errMetric.Points[0].Value).To(BeEquivalentTo(0))
+		Consistently(f).Should(Equal(0))
```
Comment (gmmeyer) [inline datadogclient/datadog_client_test.go]: Why did you change Eventually to Consistently?

File: testhelpers/fake_firehose.go [hunk 2]
```diff
@@ -100,4 +100,9 @@ func (f *FakeFirehose) ServeHTTP(rw http.ResponseWriter, r *http.Request) {
 			panic(err)
 		}
 	}
+
+	// wait a bit before closing the connection with the nozzle
+	// and forcing it to call PostMetrics
+	// this gives the nozzle time to process the envelopes we just sent
+	time.Sleep(500 * time.Millisecond)
```
Comment (gmmeyer) [inline testhelpers/fake_firehose.go]: 🍰 this is a good fix--no need to overengineer the test

File: datadogclient/datadog_client_test.go [hunk 3]
```diff
@@ -175,623 +153,237 @@ var _ = Describe("DatadogClient", func() {
...
-		err = json.Unmarshal(bodies[1], &payload)
-		Expect(err).NotTo(HaveOccurred())
-		Expect(payload.Series).To(HaveLen(3))
-
-		validateMetrics(payload, 2, 5)
+		Expect(tag1Found).To(BeTrue())
+		Expect(tag2Found).To(BeTrue())
 	})
 
-	It("sends a value 1 for the slowConsumerAlert metric when consumer error is set", func() {
```
Comment (gmmeyer) [inline datadogclient/datadog_client_test.go]: I can't tell if you removed this test or not? But I think it's an important test of the functionality that I can't find.

File: datadogclient/datadog_client_test.go [hunk 4]
```diff
@@ -175,623 +153,237 @@ var _ = Describe("DatadogClient", func() {
...
+			},
+			metrics.Point{
+				Timestamp: 2,
+				Value:     76.0,
+			},
+		}))
+		Expect(metric.Tags).To(Equal(defaultTags))
 	})
 
-	It("sends a value 0 for the slowConsumerAlert metric when consumer error is not set", func() {
```
Comment (gmmeyer) [inline datadogclient/datadog_client_test.go]: ditto on this I think this is an important test, this is the internal nozzle metric that says that the nozzle is misbehaving or behaving. It should be retained if possible

File: datadogclient/datadog_client_test.go (hunk 3 above)
Comment (ISauve) [reply datadogclient/datadog_client_test.go]: The `slowConsumerAlert` metric was moved so that the nozzle itself owns it (so as to be able to add it to the metrics map prior to sending the map to the `Client` for posting), so now https://github.com/DataDog/datadog-firehose-nozzle/blob/2660d835d1b7d5fd7e21ba094ea9b2fc8c214a28/datadogfirehosenozzle/datadog_firehose_nozzle_test.go#L160 is responsible for checking it - which it does here: https://github.com/DataDog/datadog-firehose-nozzle/blob/2660d835d1b7d5fd7e21ba094ea9b2fc8c214a28/datadogfirehosenozzle/datadog_firehose_nozzle_test.go#L190

I could definitely add another test that explicitly checks that the value gets set when `nozzle.AlertSlowConsumerError()` gets called - I didn't do it originally for fear of redundancy, but perhaps it would be worth doing a separate test specifically for it?

File: datadogclient/datadog_client_test.go (hunk 4 above)
Comment (ISauve) [reply datadogclient/datadog_client_test.go]: Similar to above, this is now covered by https://github.com/DataDog/datadog-firehose-nozzle/blob/2660d835d1b7d5fd7e21ba094ea9b2fc8c214a28/datadogfirehosenozzle/datadog_firehose_nozzle_test.go#L198

But again, could be separated into it's own test.

File: datadogclient/datadog_client_test.go (hunk 1 above)
Comment (ISauve) [reply datadogclient/datadog_client_test.go]: I actually didn't change this, github's way of lining up the changes that were made just makes it look like I did.
Here's the original: https://github.com/DataDog/datadog-firehose-nozzle/blob/master/datadogclient/datadog_client_test.go#L547
And here's the new one: https://github.com/DataDog/datadog-firehose-nozzle/blob/2660d835d1b7d5fd7e21ba094ea9b2fc8c214a28/datadogclient/datadog_client_test.go#L361

File: testhelpers/fake_firehose.go (hunk 2 above)
Comment (ISauve) [reply testhelpers/fake_firehose.go]: Fair enough!

File: datadogclient/datadog_client_test.go (hunk 4 above)
Comment (gmmeyer) [reply datadogclient/datadog_client_test.go]: That's fine!

File: datadogclient/datadog_client_test.go (hunk 1 above)
Comment (gmmeyer) [reply datadogclient/datadog_client_test.go]: Okay that's fine!

File: datadogclient/datadog_client_test.go (hunk 3 above)
Comment (gmmeyer) [reply datadogclient/datadog_client_test.go]: Ah I see, okay let's leave it. I noticed that it was there, but I missed one of the tests.
